src/test/java/teammates/test/cases/logic/CoursesLogicTest.java

```diff
@@ -438,12 +438,22 @@ private void testGetCoursesForStudentAccount() throws Exception {
         assertEquals(2, courseList.size());
 
         CourseAttributes course1 = dataBundle.courses.get("typicalCourse1");
-        assertEquals(course1.getId(), courseList.get(0).getId());
-        assertEquals(course1.getName(), courseList.get(0).getName());
 
         CourseAttributes course2 = dataBundle.courses.get("typicalCourse2");
-        assertEquals(course2.getId(), courseList.get(1).getId());
-        assertEquals(course2.getName(), courseList.get(1).getName());
+
+        boolean isCourse1Present = false;
+        boolean isCourse2Present = false;
+
+        for (CourseAttributes course : courseList) {
+            if (course.getId().equals(course1.getId()) && course.getName().equals(course1.getName())) {
+                isCourse1Present = true;
+            } else if (course.getId().equals(course2.getId()) && course.getName().equals(course2.getName())) {
+                isCourse2Present = true;
+            }
+        }
+
+        assertTrue(isCourse1Present);
+        assertTrue(isCourse2Present);
```

Comment:
Which one of these are we doing here?
1. Given list1 and list2, confirm both has the same objects irrespective of the order
2. Given list1 and list2, confirm list1 contains all objects in list2?
In either case, may be write a general utility method is the better approach? e.g. `areSameElements(List<T>, List<T>)`. The current one is just a stop gap solution that does not scale well when the number of objects involved are higher.

Another approach to consider is to avoid the problem altogether by getting rid of the non-determinism. e.g. modify the API to return the values in a specified sort order.

What do you guys think?